jetson/nav/auton_arm/stateMachine.cpp

```diff
@@ -0,0 +1,219 @@
+            if( nextState != mPhoebe->roverStatus().currentState() )
+            {
+                mPhoebe->roverStatus().currentState() = nextState;
+                mStateChanged = true;
+            }
+            return;
+        }
+        switch( mPhoebe->roverStatus().currentState() )
+        {
+            case AutonArmState::Off:
+            {
+                nextState = executeOff();
+                break;
+            }
+
+            case AutonArmState::Done:
+            {
+                nextState = executeDone();
+                break;
+            }
+
+            case AutonArmState::WaitingForTag:
+            {
+                nextState = executeWaitingForTag();
+                break;
+            }
+
+            case AutonArmState::EvaluateTag: 
+            {
+                nextState = executeEvaluateTag();
+                break;
+            }
+
+            case AutonArmState::PauseTag:
+            {
+                nextState = executePauseTag();
+                break;
+            }
+
+            case AutonArmState::SendCoordinates:
+            {
+                nextState = executeSendCoordinates();
+                break;
+            }
+
+            case AutonArmState::Unknown:
+            {
+                cerr << "Entered unknown state.\n";
+                exit(1);
+            }
+        } // switch
+        if( nextState != mPhoebe->roverStatus().currentState() )
+        {
+            mStateChanged = true;
+            mPhoebe->roverStatus().currentState() = nextState;
+        }
+        cerr << flush;
+    } // if
+} //run()
+
+// Updates the auton state (on/off) of the rover's status.
+void AutonArmStateMachine::updateRoverStatus( AutonState autonState ) {
+    mNewRoverStatus.autonState() = autonState;
+} //updateRoverStatus(AutonState autonState)
+
+// Updates the target information of the rover's status.
+void AutonArmStateMachine::updateRoverStatus( TargetPositionList targetList ) {
+    if( mPhoebe->roverStatus().currentState() != AutonArmState::WaitingForTag ) return;
+    cout << "Target received!" << endl;
+    
+    mNewRoverStatus.targetList() = targetList;
+    isTagReceived = true;
+} //updateRoverStatus(Target target)
+
+bool AutonArmStateMachine::isRoverReady() const {
+    return mStateChanged || // internal data has changed
+           mPhoebe->updateRover( mNewRoverStatus ) || // external data has changed
+           isTagReceived || // has a new tag been received 
+           mPhoebe->roverStatus().currentState() == AutonArmState::PauseTag; // if timer is active
+} //isRoverReady()
+
+void AutonArmStateMachine::publishNavState() const
+{
+    //TODO implement if we want 
+} // publishNavState()
+
+AutonArmState AutonArmStateMachine::executeOff() { 
+    if( mPhoebe->roverStatus().autonState().is_auton )
+    {
+        cout << "Reached execute off" << endl;
+        //TODO add what we want here
+    }
+    return AutonArmState::Off;
```

Comment:
this function will need a stop() function to stop all controls to the arm, maybe mPhoebe->arm->stop(), to guarantee that teleop does not continue receiving commands to move. 

alternatively teleop might check if this auton program is publishing on/off, but our publish function is empty